lib/form/js/repeatable.js

```diff
@@ -0,0 +1,266 @@
+
+  // Splits the string and pops the last element
+  var splitPop = function pop(input, splitter) {
+    var blocks = input.split(splitter);
+    blocks.pop();
+    blocks = blocks.join(splitter);
+
+    return blocks;
+  };
+
+  // Iterates through every instance of input-plugins
+  Object.keys(allSettings).forEach(function repeatableHandler(input) {
+    var idAdd = input + '--add',
+        idDelete = input + '--delete',
+        deleteHandler,
+        addHandler,
+        addButton;
+
+    deleteHandler = function deleteEventHandler(e) {
+      var node = e.currentTarget.parentNode,
+          siblingNode = node.nextElementSibling,
+          splitChild,
+          nodeId,
+          container,
+          name,
+          childNodes = node.querySelectorAll('*'),
+          parent = node.parentNode,
+          deleteNode;
+
+      // update allInputs and allIds;
+      childNodes.forEach(function deleteIDandInput(child) {
+        if (allInputs.hasOwnProperty(child.id)) {
+          delete allInputs[child.id];
+        }
+        if (allIDs.hasOwnProperty(child.id)) {
+          delete allIDs[child.id];
+        }
+      });
+
+      parent.removeChild(node);
+      allRepeatables[input].length--;
+
+      deleteNode = function deleteElement(key) {
+        var index,
+            oldChildId,
+            childName,
+            htmlFor,
+            child = key;
+        if (child.id && child.id !== '') {
+          splitChild = child.id.split('--');
+          index = splitChild[splitChild.length - 1];
+          if (!isNaN(index)) {
+            splitChild[splitChild.length - 1] = index - 1;
+            oldChildId = child.id;
+            child.id = splitChild.join('--');
+
+            // Maintains state of allInputs
+            if (allInputs.hasOwnProperty(oldChildId)) {
+              allInputs[child.id] = allInputs[oldChildId];
+              delete allInputs[oldChildId];
+            }
+            if (allIDs.hasOwnProperty(oldChildId)) {
+              // Maintains state of allIDs for changing inputs
+              allIDs[child.id] = allIDs[oldChildId];
+              delete allIDs[oldChildId];
+            }
+          }
+          childName = child.getAttribute('name');
+          if (childName) {
+            childName = childName.split('--');
+            index = childName[childName.length - 1];
+            if (!isNaN(index)) {
+              childName[childName.length - 1] = index - 1;
+              child.setAttribute('name', childName.join('--'));
+            }
+          }
+        }
+        else if (child.htmlFor && child.htmlFor !== '') {
+          splitChild = child.htmlFor.split('--');
+          htmlFor = splitChild[splitChild.length - 1];
+          if (!isNaN(htmlFor)) {
+            splitChild[splitChild.length - 1] = htmlFor - 1;
+            child.htmlFor = splitChild.join('--');
+          }
+        }
+      };
+
+      // change ids and names of subsequest instances
+      while (siblingNode) {
```

Comment:
Please group function definitions together before writing the code external to the function that consumes said functions